tooling/maven/camel-api-component-maven-plugin/src/main/java/org/apache/camel/maven/AbstractApiMethodGeneratorMojo.java

```diff
@@ -344,9 +344,15 @@ public String getCanonicalName(ApiMethodArg argument) throws MojoExecutionExcept
                         parameterizedType.append(
                             getCanonicalName(getProjectClassLoader().loadClass("java.lang." + argType)));
                     } catch (ClassNotFoundException e1) {
-                        log.warn("Ignoring type parameters <" + typeArgs + "> for argument " + argument.getName()
-                                 + ", unable to load parametric type argument " + argType, e1);
-                        ignore = true;
+                        // if the length of the artType is 1, we think that it's variable type parameter (like T in List<T>)
+                        // not perfect solution, but should work in most of the cases
+                        if (argType.trim().length() > 1) {
```

Comment:
@manonthegithub I'm not sure, but we could try to implement more elegant solution, like cast to (T) and catch ClassCastException?

Reply:
@dmvolod it is already checked that the clas does not exits on the line 335 and then 344, how would additional check help?

Reply:
Didn't do much of class casting, but I think it's impossible to cast having only string value of a class name.

Reply:
* I mean a String in runtime with the name of type arg

Reply:
Yes, you are right, but looks like not handles all generics variants. But may be it's enough for current cases.

Reply:
That is true, but it's not really possible to distinguish here between Generic parameter and nonexistent class.